content/en/cosign/keyless.md

```diff
@@ -0,0 +1,177 @@
+```
+
+That's it! No keys!
+The rest of the flags (annotations, claims, tlog, etc.) should all work the same.
+
+## Overview
+
+This uses ephemeral keys and certificates, which are signed automatically by the `fulcio` root CA.
+Signatures are stored in the `rekor` transparency log, which automatically provides an attestation
+as to when the signature was created.
+
+Information on the `fulcio` root CA can be found in the [fulcio repository](https://github.com/sigstore/fulcio).
+
+### Keys
+
+The root CA keys are hard-coded in Cosign today.
+They can only be changed by recompiling the binary.
+This will be made more configurable in the future.
+
+### OAuth Flows
+
+Cosign supports two OAuth flows today: the standard flow and the device flow.
+
+When there is no terminal attached (non-interactive mode), Cosign will automatically use the device flow
+where a link is printed to stdout.
+This link must be opened in a browser to complete the flow.
+
+### Identity Tokens
+
+In automated environments, Cosign also supports directly using OIDC Identity Tokens from specific issuers.
+These can be supplied on the command line with the `--identity-token` flag.
+The `audiences` field must contain `sigstore`.
+
+Cosign also has support for detecting some of these automated environments
+and producing an identity token.  Currently this supports Google and GitHub.
+
+#### On GCP
+
+From a GCE VM, you can use the VM's service account identity to sign an image:
+
+```shell
+$ IDENTITY_TOKEN=$(gcloud auth print-identity-token --audiences=sigstore)
+$ cosign sign --identity-token=$IDENTITY_TOKEN $IMAGE_DIGEST
+```
+
+From outside a GCE VM, you can impersonate a GCP IAM service account to sign an image:
+
+```shell
+$ IDENTITY_TOKEN=$(gcloud auth print-identity-token \
+        --audiences=sigstore \
+        --include-email \
+        --impersonate-service-account my-sa@my-project.iam.gserviceaccount.com)
+$ cosign sign --identity-token=$IDENTITY_TOKEN $IMAGE_DIGEST
+```
+
+In order to impersonate an IAM service account, your account must have the
+`roles/iam.serviceAccountTokenCreator` role.
+
+**Note**: On Google Cloud Build, standard identity tokens are not supported through the GCE metadata server.
+Cosign has a special flow for this case, where you can instruct the Cloud Build service account to impersonate
+another service account.
+To configure this flow:
+
+1. Create a service account to use for signatures (the email address will be present in the certificate subject).
+2. Grant the Cloud Build service account the `roles/iam.serviceAccountTokenCreator` role for this target account.
+3. Set the `GOOGLE_SERVICE_ACCOUNT_NAME` environment variable to the name of the target account in your cloudbuild.yaml
+4. Sign images in GCB, without keys!
+
+### Timestamps
+
```

Comment:
There needs to be a lot more here imo, so can we add a TODO about improving timestamping documentation?